### PR TITLE
Controllers

### PR DESCRIPTION
**Pull request by AlyAtalla**
For each URL  create:

A route.
An action in the correct controller.
A view file.
View files should include only placeholders written in plain HTML.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+class UsersController < ApplicationController
+end
```
Comment (V-Blaze) [inline app/controllers/users_controller.rb]: - You have done great so far :+1: :+1: But there are a few requirements you need to meet before moving to the next milestone. :+1: :100: 

- - Screenshot:
Selection_020


- [ ] As stated in the project requirement, The routes index and show` need to have a corresponding action in the controller. Currently, you do not have any action in the controller, I'll suggest you add the `index and show` action as required. Here is a snippet of how it should look like for index action. :+1: 
```ruby
def index; end
```  

- [ ] Also you have only the `index` view file. I'll suggest you add the user's `show` view file too as required. :+1: :100:

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+class PostsController < ApplicationController
+end
```
Comment (V-Blaze) [inline app/controllers/posts_controller.rb]: - [ ] Just as I pointed out above in the `UsersController`, you need to add a corresponding action and view for the routes. I'll suggest you add the `index and show` action in this controller and also add the `show` view in the view files as you currently have only the index. :+1: :smile: :small_airplane:

File: config/routes.rb [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
-  # Can be used by load balancers and uptime monitors to verify that the app is live.
+  # Health check route
   get 'up' => 'rails/health#show', as: :rails_health_check
 
-  # Defines the root path route ("/")
-  # root "posts#index"
+  resources :users
+resources :comments
+resources :likes
+resources :posts
```
Comment (V-Blaze) [inline config/routes.rb]: - Screenshot 

routes


- [ ] You have done well implementing the routes, but for your routes to look like the one in the requirement screenshot, you need to wrap the `:posts` resources in the `users resources. 
- [ ] Also according to the requirement, you only currently need the `index and show` routes. :+1: :small_airplane:  Here a snippet of how your routes should be implemented:
```ruby

  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end


```

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @AlyAtalla 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

giphy


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Added `users_controller` and `posts_controller`✔
-  Correct git flow👋 
-  Added some required view files✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tugsem — APPROVED** [review]
## STATUS: APPROVED 👍🎯  
Hello @AlyAtalla 👋 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⭐ 
✔️ correct use of gitflow
✔️routes.rb  file are corrected
✔️required actions and view files are added
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**